### ISSUE TITLE
Import from keras wrong rank array

### ISSUE THREAD

**Issue by JayadevHaddadi** [deeplearning4j/deeplearning4j#4451]
I import a keras model for mnist and i get this error: 

DL4JInvalidInputException: Got rank 2 array as input to ConvolutionLayer (layer name = convolution2d_2, layer index = 0) with shape [1, 784]. Expected rank 4 array with shape [minibatchSize, layerInputDepth, inputHeight, inputWidth]. (Wrong input type (see InputType.convolutionalFlat()) or wrong data type?) (layer name: convolution2d_2, layer index: 0)

**Comment by AlexDBlack** [comment]
@JayadevHaddadi we'll need more information than this. Can you show:
(a) the model you are importing, and
(b) the data pipeline you are using

**Comment by JayadevHaddadi** [comment]
a) https://github.com/hwchong/MNIST_DRAW/blob/master/Training/models/mnist_cnn_keras_122.h5

b) we scan for a digit in paper
resize it to 28x28x1
conv = convert to double array 784 in size
INDArray nd = Nd4j.create(conv);    // to fix the problem is to just reshape at this point or?
INDArray results = model.output(nd);

**Comment by maxpumperla** [comment]
let me have a look next week!

**Comment by AlexDBlack** [comment]
@JayadevHaddadi yes, you may be able to simply reshape (.reshape('c', new int[]{1,1,28,28}) here.
Or equivalently just add a CnnToFeedForwardPreProcessor to your network - https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/preprocessor/FeedForwardToCnnPreProcessor.java

I'll leave this open for @maxpumperla to decide if/when to close (I'm not sure if Keras actually expects a 2d input or not here)

**Comment by JayadevHaddadi** [comment]
yeah it worked fine with just that reshape. thanks.
still the keras model was not so good. only 93% acc on my validation set...
can not find any other keras models so i'll just keep on trying to train a better one directly in dl4j
thanks
